src/geometrycollection.cpp

```diff
@@ -21,6 +21,34 @@ void GeometryCollection::clear()
     }
 }
 
+int GeometryCollection::findMatchingRow(const std::string & filename_substr)
+{
+    for (unsigned row = 0; row < m_geometries.size(); row++)
+    {
+		if (std::string::npos != m_geometries[row]->fileName().toStdString().find(filename_substr))
+        {
+            printf("MATCHED '%s'\n", m_geometries[row]->fileName().toStdString().c_str());
```

Comment:
Use `tfm::printf` and you won't need the `c_str()` ;-)

Regardless of that, let's remove the printf debug logging before merging this.